### PR TITLE
update megaman to version 0.2 release

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

**Comment by jakevdp** [comment]
Strange... CI is failing saying it can't download the package, but I have no problem locally:
```
$ curl -O https://pypi.python.org/packages/source/m/megaman/megaman-0.2.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   306  100   306    0     0   1721      0 --:--:-- --:--:-- --:--:--  2217
$ ls
megaman-0.2.tar.gz
```
Any ideas why this might be happening?

**Comment by jakevdp** [comment]
Ah – turns out I was downloading the 404 page :grin:

**Comment by jakevdp** [comment]
Ah – PyPI has a new URL scheme: http://lists.science.uu.nl/pipermail/nix-dev/2016-April/020237.html

Found a conda-forge discussion here: https://github.com/conda-forge/staged-recipes/issues/481

**Comment by jakevdp** [comment]
Travis is failing with
```
dyld: Library not loaded: @rpath/libisl.10.dylib
```

**Comment by ocefpaf** [comment]
> Travis is failing with
```
dyld: Library not loaded: @rpath/libisl.10.dylib
```

Not sure what is going on, but did you try to unpin `gcc` and `libgcc`?

**Comment by jakevdp** [comment]
They're currently pinned to 4.8* on osx. Should that be changed?

**Comment by ocefpaf** [comment]
> They're currently pinned to 4.8* on osx. Should that be changed?

It is worth a try :smile:

**Comment by jakevdp** [comment]
Still failing with the same error 😡  I'm not sure what to do...

What in the OSX build has changed since when this previously worked?

**Comment by ocefpaf** [comment]
Any ideas @msarahan?

**Comment by ocefpaf** [comment]
@jakevdp can you try adding (pinning) `isl` like [this](https://github.com/conda-forge/staged-recipes/pull/810/commits/1c3e8668495085e8da73700e7858ffd45d5fcf80). That will ensure you'll get the default channel `isl` instead of `conda-forge`'s. I believe ours might be broken.

Note that only on `OS X` `conda` is pulling `conda-forge`'s `isl`... That deserves some investigation.

**Comment by jakevdp** [comment]
Looks like it worked!

One question: I un-pinned gcc here; I recall that being recommended for some reason previously. Also #1 mentions gcc should be a dependency on Linux as well... any opinions on those?

**Comment by ocefpaf** [comment]
@jakevdp that seems to work. Can you please open an issue` reporting our faulty `isl`?

https://github.com/conda-forge/isl-feedstock/issues

**Comment by frol** [comment]
There is nothing wrong with ISL package on conda-forge. GCC 4.8, which is in the default anaconda channel, supports only ISL 0.12.2. GCC 5, for example, supports 0.13 .. 0.14, and GCC - 0.14 .. 0.16. It is just a poor GCC package on anaconda, which doesn't pin supported versions of ISL at all, as well as cloog, mpc, and mpfr.

**Comment by jakevdp** [comment]
Ah, thanks @frol. So is the best course for the time-being to pin gcc 4.8* and isl 0.12* in these recipes?

**Comment by frol** [comment]
@jakevdp As far as I can tell, yes - https://github.com/conda-forge/staged-recipes/issues/436. However, using `gcc` dependency is controversial - https://github.com/conda-forge/conda-forge.github.io/issues/29.

**Comment by jakevdp** [comment]
In this case, we're depending on gcc in order to use OpenMP.

**Comment by ocefpaf** [comment]
@jakevdp no need to pin `gcc` as that won't ensure you'll get the right `isl`. That problem only exist b/c `conda-forge` added a newer `isl`, so only `isl` needs to be pinned. (Pinning `gcc` here won't hurt though.)

> However, using `gcc` dependency is controversial

Indeed. But note that since `numpy` with `openblas` was added `libgcc` will be present in a number of projects causing all sort of havoc.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,7 +19,8 @@ requirements:
         - numpy x.x
         - cython
         - flann
-        - gcc 4.8*  # [osx]
+        - isl 0.12.*   # [osx]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Add this to run time too to ensure it won't get updated when people install this with other software.

**Comment by frol** [comment]
It is better to pin GCC when ISL is pinned. Otherwise, once GCC will be upgraded, you will get some weird compilation errors instead of dependency error.

**Comment by ocefpaf** [comment]
> It is better to pin GCC when ISL is pinned.

Unless you pin it down to a release there is nothing ensuring that this would work. The current package on `defaults` for `gcc` should pin `isl` to avoid this. Otherwise the mismatch can always happen in updates/changes.

File: recipe/meta.yaml (hunk 1 above)
Comment (jakevdp) [inline recipe/meta.yaml]: sounds good

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,14 +19,16 @@ requirements:
...
+        - gcc 4.8*     # [osx]
     run:
         - python
         - numpy x.x
         - scipy >=0.16
         - scikit-learn >=0.17
         - pyamg
         - pyflann
+        - isl 0.12.*   # [osx]
         - libgcc 4.8*  # [osx]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: If you unpin `gcc` you need to unpin `libgcc` too :smile_cat:  (Sorry it seems almost like I am putting you on a fools errands on purpose, but I assure you I am just a bad reviewer :stuck_out_tongue_winking_eye::)

Comment (jakevdp) [inline recipe/meta.yaml]: I plan on leaving them both pinned, I think.

Comment (ocefpaf) [inline recipe/meta.yaml]: > I plan on leaving them both pinned, I think.

That is fine.

**Comment by jakevdp** [comment]
Looks like tests are on their way to passing.

I'm going to be away from my computer this evening, but getting this merged is a bit time-sensitive since we're re-submitting a paper for review with install instructions based on conda-forge. Can someone merge for me ASAP if things look to be in order? Thanks!

**Comment by ocefpaf** [comment]
@jakevdp I saw that the tests passed before so let's be naughty and merge this before Travis-CI finishes :wink:

**Pull request merged**